### ISSUE TITLE
Documenting your API: add highlighted chunk of code in docstrings 

### ISSUE THREAD

**Issue by nastasi-oq** [encode/django-rest-framework#5374]
Thinking that also highlighted code-blocks were enabled in the dynamic documentation I started to investigate why code-block (also without highlight) not works and at the end I found that there is a strip that remove any kind of indentation, that is bad for code-blocks.
```
--- schemas_orig.py	2017-08-30 14:45:38.061047196 +0000
+++ schemas.py	2017-08-30 14:46:10.317254482 +0000
@@ -476,7 +476,7 @@
             return formatting.dedent(smart_text(method_docstring))
 
         description = view.get_view_description()
-        lines = [line.strip() for line in description.splitlines()]
+        lines = [line for line in description.splitlines()]
         current_section = ''
         sections = {'': ''}
```
At the end, adding this change and a proper Preprocessor for markdown in the ``render_markdown``  function I achieved my goal:
![drf-new-feature](https://user-images.githubusercontent.com/1670278/29878850-dcd0de54-8da3-11e7-833b-7d3f4b66741c.png)

Starting from this docstring:
```
    """
get:
Return the list of country objects with **iso2**,
**name** and **region** fields.

@@ json @@
[
    {
        "first": "string-value",
        "second": null
    },
    {
        "one": 1,
        "two": 2
    }
]
@@

"""
```
I want to submit a PR with this (small) change but before I would know if the ``line.strip()`` has a precise meaning and cannot be removed or not.
Thank you, Matteo.

**Comment by carltongibson** [comment]
Related #5240. 

Yes. I’d be happy to see something here. 

The strip is likely just an error. (See linked issue)

**Comment by nastasi-oq** [comment]
First incomplete iteration of the PR, if you confirm that the missing parts list is complete I will continue to work on that and add the rest of code.
https://github.com/encode/django-rest-framework/pull/5377

**Comment by carltongibson** [comment]
Hey @nastasi-oq, if you wanted to have another go at #5377, we're rolling the 3.7.0 release now. This would be a nice addition. 👍

**Comment by nastasi-oq** [comment]
I updated just this morning my codebase with the latest master and added the current code, tests and optionality are still missing and I want to work on them in the next days.

**Comment by carltongibson** [comment]
Awesome. Great. 

If you’d like input, open a fresh PR and we can discuss. 

👍🏽

**Comment by nastasi-oq** [comment]
@carltongibson question: markdown package is considered mandatory or optional ?

**Comment by carltongibson** [comment]
It's optional. 

I think you'll need to reorder pygments and markdown in compat, and then conditionally define your extra function inside the markdown block. (If I remember right... 😳)

**Comment by nastasi-oq** [comment]
@carltongibson: this is the new PR: https://github.com/encode/django-rest-framework/pull/5462
If the code is OK for you I proceed with tests.

**Comment by carltongibson** [comment]
Looks good so far, yes. 👍🏽

**Comment by carltongibson** [comment]
Closed by #5462
